### ISSUE TITLE
[Idea] Emulation of Goodwee meter to use other meters with Goodwee inverter.

### ISSUE THREAD

**Issue by PovilasID** [smlx/goodwe-exporters#87]
Hey,

TL;DR Do you think it is possible to use utilities smart meter data to emulate the Goodwee smart meter using cheap ESP32 board Lilygo T-CAN485?

My utilities company has installed a smart meter that offers a P1 protocol connection to monitor output/consumption and you can get cheap dongle from [homewizard](https://www.homewizard.com/p1-meter/) to get the data into Home Assistant and from there anywhere else but I am not sure how to feed back to inverter.
My objective is to make sure that the grid output is capped to 10kW (my service return limit) and if my local usage is lets say 2kW that the inverter cap is raised to 12 kW. This way grid keeps getting 10kW I get 2 kW.
My idea would be to feed data from P1 meter to something like [Lilygo T-CAN485](https://www.lilygo.cc/products/t-can485) that would emulate the smart meter and save ~100 USD on a redundant smart meter.

**Comment by smlx** [comment]
> Hey,

Hi!

> TL;DR Do you think it is possible to use utilities smart meter data to emulate the Goodwee smart meter using cheap ESP32 board Lilygo T-CAN485?

Yes, this might be possible if the utilities smart meter provides similar data to the Goodwe smart meter.

> My utilities company has installed a smart meter that offers a P1 protocol connection to monitor output/consumption and you can get cheap dongle from [homewizard](https://www.homewizard.com/p1-meter/) to get the data into Home Assistant and from there anywhere else but I am not sure how to feed back to inverter.

You have a GoodWe inverter? What do you mean by "feed back to inverter"?

> My objective is to make sure that the grid output is capped to 10kW (my service return limit) and if my local usage is lets say 2kW that the inverter cap is raised to 12 kW. This way grid keeps getting 10kW I get 2 kW. My idea would be to feed data from P1 meter to something like [Lilygo T-CAN485](https://www.lilygo.cc/products/t-can485) that would emulate the smart meter and save ~100 USD on a redundant smart meter.

I don't quite understand your use case, but emulating a GoodWe HK1000 should be possible.

**Comment by PovilasID** [comment]
Yah it dose provide a similar data. 
I was looking into buying inverters and as I was doing my research saw that I may save a bit of money by skipping the smart meter but since then I was able to find a supplier with a well priced inverter and meter bundle. 

It was not entirely clear but my service will have two smart meters. The utilities meter is also smart it just incompatible with the inverter. My array will grow to be larger than 10kW, so dynamically controlling the cap  to both supply the gird and my usage at max utilization the same time would be very nice.

I gather that is not in the plans, so I think we can close this on?
